pkg/controller/jobframework/interface.go

```diff
@@ -36,11 +36,21 @@ import (
 	"sigs.k8s.io/kueue/pkg/util/maps"
 )
 
-// GenericJob if the interface which needs to be implemented by all jobs
+// BaseJob is the interface which needs to be implemented by all jobs
 // managed by the kueue's jobframework.
-type GenericJob interface {
+type BaseJob interface {
```

Comment:
I would call it `GenericWorkload` tbh. Since it is not necessarily as Job, as but could be serving too. I imagine we will have 

```
              GenericWorkload
         /                                  \
     GenericJob                   ServingWorkload
```
     
in the future

Reply:
As suggested in other comments - let's try the approach without the new interface.

Reply:
Done